src/proxy/pool.rs

```diff
@@ -996,10 +1001,35 @@ mod test {
             ..crate::config::parse_config().unwrap()
         };
         let sock_fact = Arc::new(crate::proxy::DefaultSocketFactory);
-        let cert_mgr = proxy::ScopedSecretManager::new(identity::mock::new_secret_manager(
-            Duration::from_secs(10),
+
+        let mut state = ProxyState::new(None);
```

Comment:
nit: probably should be a series of helpers/we may already have a helper that does some or all of this.